boards/riscv/esp32c3_devkitm/esp32c3_devkitm.yaml

```diff
@@ -3,4 +3,4 @@ name: ESP32-C3
 type: mcu
 arch: riscv
 toolchain:
-  - espressif
+  - zephyr
```

Comment:
@stephanosio Can we proceed merging this to have new toolchain available and then we handle CI stuff in a dedicated matter? I can restore this entry to `espressif` and modify the documentation of make sure `export ZEPHYR_TOOLCHAIN_VARIANT="zephyr"` is kept. What do you think?

Reply:
Doing so will make the process of release QA testing rather inconvenient/prone to human errors (unless someone explicitly remembers to restore this patch locally while testing).

Since it is important that we keep building these platforms in the CI so that they do not bit-rot away, I am in favour of adding `west espressif update` in the twister workflow with a FIXME comment and an issue (bug) to keep track of this.

Reply:
@sylvioalves Let's discuss this in today's dev-review or next week's TSC. If you are available today to join the dev-review meeting, please ping Maureen so she can add it to the agenda.

Reply:
I think dev-review was cancelled this week. So only next week's TSC. Anyway, I thought about merging this to enable sdk toolchain but have another PR enabling CI integration.